spec/controllers/course/assessment/submission/submissions_controller_spec.rb

```diff
@@ -80,12 +81,12 @@
 
     describe '#reattempt_question' do
       let(:submission) do
-        create(:course_assessment_submission, assessment: assessment, creator: user)
+        create(:course_assessment_submission, assessment: assessment2, creator: user)
```

Comment:
Actually maybe the fix to this is not to create `assessment2` here, but to only lazy load `immutable_sumbission`. Can help check if changing line 12 from `let!(:immutable_submission)` to `let(:immutable_submission)` fixes it instead?